### ISSUE TITLE
MeasureValidator is incorrectly requiring subjectList count to match population count

### ISSUE THREAD

**Issue by brynrhodes** [hapifhir/org.hl7.fhir.core#1378]
For non-patient-based measures, the population count will not match the subject count (subject count will be per subject, where the population counts will be the items in the population such as encounters or procedures).

This is where the logic in the MeasureValidator is:
https://github.com/hapifhir/org.hl7.fhir.core/blob/96b5b6c5205db4b7140fe04d437cb8e25f81b571/org.hl7.fhir.validation/src/main/java/org/hl7/fhir/validation/instance/type/MeasureValidator.java#L504

Basically, this rule is only true for patient-based measures (i.e. measures where the populationBasis is Patient)

**Comment by grahamegrieve** [comment]
have you got an example to add to the validator test cases?
